Sources/Controllers/MessagesViewController.swift

```diff
@@ -59,6 +61,18 @@ UICollectionViewDelegateFlowLayout, UICollectionViewDataSource {
         return false
     }
 
+    /// A CGFloat value that adds to (or, if negative, subtracts from) the automatically
+    /// computed value of `messagesCollectionView.contentInset.bottom`. Meant to be used
+    /// as a measure of last resort when the built-in algorithm does not produce the right
+    /// value for your app. Please let us know when you end up having to use this property.
+    open var extraBottomInset: CGFloat = 0 {
+        didSet {
+            let delta = extraBottomInset - oldValue
+            guard abs(delta) > coordinatePrecision else { return }
```

Comment:
@andreyvit Seems like you are very careful to the comparison of `double` type. 😂 TBO these things maybe not needed, I think `1e-6` is only applied to compare the equality of two `double`, in our situations, we don't need accurate precision, just get the delta and assign to the `messageCollectionViewBottomInset`.